src/Plugin/InvokeStrategy/OnEventStrategy.php

```diff
@@ -22,4 +26,24 @@ public function invoke($handler, $message): void
     {
         $handler->onEvent($message);
     }
+
+    public function attachToMessageBus(MessageBus $messageBus): void
+    {
+        $this->listenerHandlers[] = $messageBus->attach(
+            MessageBus::EVENT_DISPATCH,
+            function (ActionEvent $actionEvent): void {
+                $message = $actionEvent->getParam(MessageBus::EVENT_PARAM_MESSAGE);
+                $handlers = $actionEvent->getParam(EventBus::EVENT_PARAM_EVENT_LISTENERS);
+
+                foreach ($handlers as $handler) {
+                    $this->invoke($handler, $message);
```

Comment:
can be simply: `$handler->onEvent($message);` and method `public function invoke($handler, $message): void` can be removed.